### PR TITLE
Intermediate.Austin.2017Q1.Rohan.Mutalik

### PR DESCRIPTION
**Pull request by rmutalik**
# Checkpoint Rubric

This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

## Checkpoint 1: Towers of Hanoi Terminal App
- [ ] All tests passed: 55 points
- [ ] Properly commented code: 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of JavaScript: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 2: Towers of Hanoi GUI App
- [ ] All tests passed: 55 points
- [ ] Properly commented code: 15 points
- [ ] Properly indented code: 15 points
- [ ] Demonstrated effective use of jQuery: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

## Checkpoint 3: Final Project Gist Blog
- [ ] Effective use of AJAX: 20 points
- [ ] Effective use of loops: 20 points
- [ ] Effective use of github API: 20 points
- [ ] Style and design: 10 points
- [ ] Properly commented code: 15 points
- [ ] Properly indented code: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

### CODE REVIEW

File: apps/02RockPaperScissors.js [hunk 1]
```diff
@@ -10,10 +10,48 @@ var rl = readline.createInterface({
 
 function rockPaperScissors(hand1, hand2) {
 
+  /*
+  Edge cases
+  */
+  // Case 1: Inputs are not completely lowercase
+  hand1 = hand1.toLowerCase();
```
Comment (jw56578) [inline apps/02RockPaperScissors.js]: Good reuse of existing variable. No need to make another one.

File: apps/02RockPaperScissors.js [hunk 2]
```diff
@@ -10,10 +10,48 @@ var rl = readline.createInterface({
 
 function rockPaperScissors(hand1, hand2) {
 
+  /*
+  Edge cases
+  */
+  // Case 1: Inputs are not completely lowercase
```
Comment (jw56578) [inline apps/02RockPaperScissors.js]: Great commenting

File: apps/02RockPaperScissors.js [hunk 3]
```diff
@@ -10,10 +10,48 @@ var rl = readline.createInterface({
...
+  hand1 = hand1.toLowerCase();
+  hand2 = hand2.toLowerCase();
+  // Case 2: Inputs are not strings
+  console.log(typeof hand1);
+  console.log(typeof hand2);
+  if((hand1 !== 'rock' && hand1 !== 'paper' && hand1 !== 'scissors')
```
Comment (jw56578) [inline apps/02RockPaperScissors.js]: Nice user input validation

File: homework/04LessonFour.js [hunk 4]
```diff
@@ -77,7 +83,7 @@ var nestedObjects = {
   }
 };
 // update the starring property to have the value 'Tim Robbins'
-
+nestedObjects['users']['bob roberts']['starring'] = 'Tim Robbins';
```
Comment (pieroapretto) [inline homework/04LessonFour.js:86]: Nicely done.

File: apps/03TicTacToe.js [hunk 5]
```diff
@@ -23,24 +28,72 @@ function printBoard() {
...
+  }
 }
 
+// Checks for any type of win
 function checkForWin() {
-  // Your code here
+  if(horizontalWin() || verticalWin() || diagonalWin()) {
+    console.log("Player " + playerTurn + " Won!");
+    resetScreen();
+    return true;
```
Comment (pieroapretto) [inline apps/03TicTacToe.js:76]: Logic checks out!

**Pull request closed without merging**